src/components/TaskDetailsSheet/logs.tsx

```diff
@@ -0,0 +1,93 @@
+import { useQuery } from "@tanstack/react-query";
+import { useLocation } from "@tanstack/react-router";
+
+import { type RunDetailParams, runDetailRoute } from "@/router";
+import { API_URL, RUNS_BASE_PATH } from "@/utils/constants";
+
+const LogDisplay = ({
+  logs,
+  isLoading,
+}: {
+  logs: {
+    log_text: string;
+  };
+  isLoading: boolean;
+}) => {
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!logs || !logs.log_text) {
+    return <div>No logs available</div>;
+  }
+
+  const lines = logs.log_text.split("\n");
+  return (
+    <div className="space-y-1">
+      {lines.map((line: string, index: number) => (
+        <div key={index}>{formatLogLine(line)}</div>
+      ))}
+    </div>
+  );
+};
+
+const formatLogLine = (line: string) => {
+  const timestampMatch = line.match(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}/);
+  if (!timestampMatch) return line;
+
+  const timestamp = new Date(timestampMatch[0]);
+  const content = line.slice(timestampMatch[0].length + 1);
+
+  return (
+    <div key={line} className="flex">
+      <span className="text-gray-400 mr-4 select-none whitespace-nowrap shrink-0">
+        {timestamp.toLocaleTimeString()}
+      </span>
+      <span className="break-all">{content}</span>
+    </div>
+  );
+};
+
+const getLogs = async (executionId: string) => {
+  const response = await fetch(`${API_URL}/api/executions/${executionId}/logs`);
+  return response.json();
+};
+
+const LogsInner = () => {
+  const { id: executionId } = runDetailRoute.useParams() as RunDetailParams;
```

Comment:
Are you sure you're using correct execution ID?

You need to be using the **child** task execution ID, not the parent graph execution ID.